Framework/CurveFitting/src/GSLFunctions.cpp

```diff
@@ -139,6 +139,7 @@ int gsl_df(const gsl_vector *x, void *params, gsl_matrix *J) {
 
   EigenMatrix m_tr = m.tr();
   std::copy(&m_tr.mutator().data()[0], &m_tr.mutator().data()[J_tr->size1 * J_tr->size2], &J->data[0]);
+  gsl_matrix_free(J_tr);
```

Comment:
J_tr would still leak if an exception is thrown on line 137. a `std::unique_ptr` with a custom deleter works well.

```
std::unique_ptr<gsl_matrix, decltype(&gsl_matrix_free)> J_tr(gsl_matrix_calloc(J->size2, J->size1), gsl_matrix_free);
```

close_file

Reply:
Changing to a `unique_ptr` is a much more elegant solution. Thank you for pointing it out.